readium/navigator-media2/src/main/java/org/readium/navigator/media2/MediaNavigator.kt

```diff
@@ -274,16 +278,15 @@ class MediaNavigator private constructor(
         Ongoing
     }
 
-    sealed class Exception : kotlin.Exception() {
+    sealed class Exception(message: String) : kotlin.Exception(message) {
 
         class SessionPlayer internal constructor(
-            internal val error: SessionPlayerError,
-            override val message: String = "${error.name} error occurred in SessionPlayer."
-        ) : Exception()
+            internal val error: SessionPlayerError
+        ) : Exception("${error.name} error occurred in SessionPlayer.")
+
+        class IllegalArgument(message: String): Exception(message)
 
-        class IllegalArgument internal constructor(
-            override val message: String
-        ): Exception()
+        class ResourceNotFound(val href: String) : Exception("Resource not found at $href")
     }
```

Comment:
`IllegalArgument` is used for the exact same purpose (stale link or locator). I think it's good enough, though it might be better named `InvalidArgument`.